### PR TITLE
perf(combobox): Avoid materializing EditableText unnecessarily

### LINKED ISSUE #15406: ComboBox eagerly materializes the EditableTextBox
### Current behavior

This line:
https://github.com/unoplatform/uno/blob/129e5b22bf76a63e775a752fab5035f612caaf01/src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs#L104

Causes the creation of a x:Load false control in the template, causing unnecessary work when editable is set to false.

### Expected behavior

Faster ComboBox

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

5.1.50

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #15406

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs [hunk 1]
```diff
@@ -72,6 +78,52 @@ public void Init()
...
+
+			// EditableText is only available in fluent style.
+			using var _ = StyleHelper.UseFluentStyles();
+			var SUT = new ComboBox();
+			await UITestHelper.Load(SUT);
+
+			Assert.IsFalse(SUT.IsEditable);
+			Assert.IsNull(_editableTextField.GetValue(SUT));
+
+			SUT.IsEditable = true;
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs:121]: I'd tend to think that this particular test is not going to work, as `_editableText` is not reevaluated to make the ComboBox work in that mode. IsEditable dose not seem to be a mode we're supporting properly at this time. I'd say that we can keep this test, but keep it ignored for now.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs:121]: @jeromelaban This test already has:

```csharp
			if (!ApiInformation.IsPropertyPresent(typeof(ComboBox), "IsEditable"))
			{
				Assert.Inconclusive();
			}
```

The intent is that it's now ignored, but once IsEditable is implemented, the test gets  "enabled" automatically.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.1

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs:121]: I missed that! Thanks!!

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.1

**Pull request merged**